back/qc_api/models/algorithm/snippet.py

```diff
@@ -17,6 +17,7 @@ class Snippet(PolymorphicModel):
         User,
         related_name='liked_snippets'
     )
+    type = models.CharField(max_length=10)
```

Comment:
Since this is restricted into four types, I think it's better to declare an enum.

Example:
```suggestion
class Snippet(PolymorphicModel):
    class SnippetType(models.TextChoices):
        SCOPE = 'SCOPE'
        BUY = 'BUY'
        SELL = 'SELL'
        AMOUNT = 'AMOUNT'
    type = models.CharField(
        max_length=6
        choices=SnippetType.choices,
        default=SnippetType.SCOPE,
    )

```

Reply:
#52